File: 프로그래머스/lv2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    static long store[];
    public long dp(int n) {
        if(n == 1) return store[n] = 1;
        if(n == 2) return store[n] = 2;
        if(store[n] == 0) {
            store[n] = (dp(n - 1) + dp(n - 2)) % 1234567;
        }
        
        return store[n];
    }
    
    public long solution(int n) {
        long answer = 0;
        store = new long[n + 1];
        answer = dp(n);
        return answer;
    }
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.js
function solution(n) {
    const store = new Array(2001).fill(0);
    store[1] = 1;
    store[2] = 2;
    
    function dp(n) {
        if(store[n] === 0) {
            store[n] = Math.floor((dp(n - 1) + dp(n - 2)) % 1234567);
        }
        
        return store[n]
    }
    
    return Math.floor(dp(n) % 1234567);
}